docs/guides/add-a-custom-index.md

```diff
@@ -0,0 +1,280 @@
+
+---
+
+### 3. Define the Indexer Object
+
+Implement a struct that represents your indexer and implements the `ProcessTransaction` trait.
+
+```rust
+pub struct MyProjIndexer {
+    start_height: u64,
+    track_inputs: bool,
+}
+
+impl ProcessTransaction for MyProjIndexer {
+    fn process_tx(
+        &mut self,
+        task: &mut IndexerTask,
+        tx: &Transaction,
+        ctx: &IndexerContext,
+    ) -> Result<(), Error> {
+        ...
+    }
+}
+```
+
+Where:
+
+-   `task`: read/write interface to storage
+-   `tx`: the transaction being processed
+-   `ctx`: context with input resolver, block height, hash, network, arbitrary data to outputs, etc.
+
+A _resolver_ lets you provide a transcation input UTXO reference and receive the corresponding transaction output UTXO.
+
+Reference:
+[`runes/indexer.rs#L41-L61`](https://github.com/maestro-org/maestro-symphony/blob/main/src/sync/stages/index/indexers/custom/runes/indexer.rs#L41-L61)
+
+---
+
+### 4. Implement and Handle Config
+
+Your indexer should expose a `new()` function that takes a configuration struct and returns an instance of the indexer.
+
+This enables configuration-driven behavior such as start_height, track_inputs, or custom logic.
+
+```rust
+impl MyProjIndexer {
+    pub fn new(config: MyProjIndexerConfig) -> Result<Self, Error> {
+        let start_height = config.start_height;
+        let track_inputs = config.track_inputs;
+
+        Ok(Self {
+            start_height,
+            track_inputs,
+        })
+    }
+}
+
+#[derive(Clone, Debug, Deserialize)]
+pub struct MyProjIndexerConfig {
+    #[serde(default)]
+    pub start_height: u64,
+    #[serde(default)]
+    pub track_inputs: bool,
+}
+```
+
+The `MyProjIndexerConfig` struct should define fields relevant to your indexer.
+
+[Reference](https://github.com/maestro-org/maestro-symphony/blob/main/src/sync/stages/index/indexers/custom/runes/indexer.rs#L41-L72)
+
+### 5. Define Storage Tables
+
+Define custom key-value tables for storing the new data using the `define_indexer_table!` macro.
+
+```
+src/sync/stages/index/indexers/custom/my_proj/tables.rs
+```
+
+Example:
+
+```rust
+define_indexer_table! {
+    name: MyProjIndexerKV,
+    key_type: ScriptPubKey,
+    value_type: u64,
+    indexer: TransactionIndexer::MyProjIndexer,
+    table: 0
+}
+```
+
+```rust
+// key-value:
+address => tx_count
+```
+
+Each table must:
+
+-   Have a unique `table` ID
+-   Use your new enum variant
+-   Use key/value types that implement `Encode` and `Decode`
+
+Reference:
+[`tx_count_by_address.rs#L20-L26`](https://github.com/maestro-org/maestro-symphony/blob/main/src/sync/stages/index/indexers/custom/tx_count_by_address.rs#L20-L26)
+
+---
+
+### 6. Implement `ProcessTransaction`
+
+Process each transaction by:
+
+-   Iterating over inputs, ouputs, resolving UTXOs, etc.
+-   Reading/writing to storage with
```

Comment:
sentence seems cut short

Reply:
Ah it was supposed to reference the below `task.get/task.put` operations